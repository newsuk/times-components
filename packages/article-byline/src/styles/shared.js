import styleguide from "@times-components/styleguide";

const { colours, fontFactory } = styleguide();
const shared = {
  link: {
    color: colours.functional.action,
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    })
  },
  text: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    }),
    color: colours.functional.secondary,
    flexDirection: "row"
  }
};

export default shared;
