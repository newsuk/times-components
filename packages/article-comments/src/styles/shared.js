import styleguide from "@times-components/styleguide";

const { colours, fontFactory, spacing } = styleguide();
const styles = {
  button: {
    marginBottom: spacing(0),
    marginTop: spacing(5),
    maxWidth: 215
  },
  container: {
    alignItems: "center",
    borderStyle: "solid",
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1,
    marginBottom: spacing(5),
    width: "100%"
  },
  headline: {
    color: colours.functional.primary,
    ...fontFactory({
      font: "headline",
      fontSize: "commentsHeadline"
    }),
    maxWidth: 315,
    paddingBottom: spacing(2),
    paddingTop: spacing(6),
    textAlign: "center"
  },
  link: {
    color: colours.functional.action
  },
  supporting: {
    ...fontFactory({
      font: "supporting",
      fontSize: "tertiary"
    }),
    color: colours.functional.secondary,
    maxWidth: 280,
    textAlign: "center"
  }
};

export default styles;
