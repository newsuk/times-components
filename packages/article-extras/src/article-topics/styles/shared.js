import { colours, tabletWidth } from "@times-components/styleguide";

const sharedStyles = {
  topicsContainer: {
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1
  },
  topicsContainerTablet: {
    alignSelf: "center",
    width: tabletWidth
  },
  topicsMetaContainer: {
    justifyContent: "flex-start"
  }
};

export default sharedStyles;
