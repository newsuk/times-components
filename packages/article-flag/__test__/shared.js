/* eslint-env jest */

import "react-native";
import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";

import ArticleFlag, {
  NewArticleFlag,
  UpdatedArticleFlag,
  ExclusiveArticleFlag,
  SponsoredArticleFlag
} from "../article-flag";

module.exports = () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ArticleFlag title="articleFlag" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders New flag correctly", () => {
    const tree = renderer.create(<NewArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Updated flag correctly", () => {
    const tree = renderer.create(<UpdatedArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Exclusive flag correctly", () => {
    const tree = renderer.create(<ExclusiveArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Sponsored flag correctly", () => {
    const tree = renderer.create(<SponsoredArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });
};
