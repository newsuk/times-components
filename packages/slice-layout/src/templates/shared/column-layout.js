import React, { Fragment } from "react";
import { View } from "react-native";
import { ItemRowSeparator } from "./index";

const ColumnLayout = ({tiles}) => (
  <View>
    {tiles.map((tile, index) => (
        <Fragment>
          {tile()}
          {index !== tiles.length - 1 ? <ItemRowSeparator /> : null}
        </Fragment>
      )
    )}
  </View>
);

export default ColumnLayout;
