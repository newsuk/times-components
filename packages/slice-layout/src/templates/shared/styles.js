import { colours, spacing } from "@times-components/styleguide";

const styles = {
  itemColSeparator: {
    borderColor: colours.functional.keyline,
    borderRightWidth: 1,
    borderStyle: "solid",
    marginVertical: spacing(2)
  },
  itemRowSeparator: {
    borderBottomWidth: 1,
    borderColor: colours.functional.keyline,
    borderStyle: "solid",
    marginHorizontal: spacing(2)
  }
};

export default styles;
