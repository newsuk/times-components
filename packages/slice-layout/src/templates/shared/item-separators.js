import React from "react";
import { TcView } from "@times-components/utils";
import PropTypes from "prop-types";
import styles from "./styles";

const ItemColSeparator = ({ style }) => (
  <TcView style={{ ...styles.itemColSeparator, ...style }} />
);
const ItemRowSeparator = ({ style }) => (
  <TcView style={{ ...styles.itemRowSeparator, ...style }} />
);

ItemColSeparator.defaultProps = {
  style: {}
};

ItemRowSeparator.defaultProps = {
  style: {}
};

ItemColSeparator.propTypes = {
  style: PropTypes.shape({})
};

ItemRowSeparator.propTypes = {
  style: PropTypes.shape({})
};

export { ItemColSeparator, ItemRowSeparator };
