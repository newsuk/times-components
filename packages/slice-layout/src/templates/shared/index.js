import { ItemColSeparator, ItemRowSeparator } from "./item-separators";

export { ItemColSeparator, ItemRowSeparator };
