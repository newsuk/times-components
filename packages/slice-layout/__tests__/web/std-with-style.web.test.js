import shared from "../std-with-style.web";

shared();
