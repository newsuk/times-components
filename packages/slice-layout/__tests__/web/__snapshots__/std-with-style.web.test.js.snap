// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. no child elements 1`] = `""`;

exports[`2. a single child element 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom-color: #DBDBDB;
  border-bottom-width: 1px;
  border-style: solid;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-left: 10px;
  padding-right: 10px;
}

.c0 .imageContainerClass {
  display: block;
}

.c0 .summaryHidden {
  display: none;
}

.c0 .summary125Class {
  display: block;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-bottom: 10px;
  padding-top: 10px;
  width: 100%;
}

@media (-webkit-min-device-pixel-ratio:2) {
  .c1 {
    border-bottom-width: 0.5px;
  }
}

@media (min-width:768px) {
  .c3 {
    -webkit-flex-basis: 0 !important;
    -ms-flex-preferred-size: 0 !important;
    flex-basis: 0 !important;
    padding-left: 0;
    padding-right: 0;
  }
}

@media (min-width:768px) {
  .c0 .imageContainerClass {
    display: block;
  }

  .c0 .headlineClass {
    font-size: 25px;
    line-height: 28px;
    margin-top: 5px;
  }
}

@media (min-width:768px) {
  .c0 .imageContainerClass {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
    margin-bottom: 0;
    min-width: auto;
    padding-right: 10px;
  }

  .c0 .contentContainerClass {
    -webkit-flex-basis: 0 !important;
    -ms-flex-preferred-size: 0 !important;
    flex-basis: 0 !important;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    min-width: 300px;
  }
}

@media (min-width:1024px) {
  .c0 .summary125Class {
    display: block;
  }

  .c0 .summary145Class {
    display: none;
  }
}

@media (min-width:768px) {
  .c2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c2 {
    width: 80.8%;
  }
}

@media (min-width:1024px) {
  .c2 {
    width: 56.2%;
  }
}

<configweb__Base>
  <div
    className="c0 configweb__Base-sc-1adgnqc-0 configweb__Base-sc-1adgnqc-1 configweb__Base-sc-1adgnqc-2 c0 css-view-1dbjc4n"
  >
    <responsiveweb__SliceContainer>
      <div
        className="c1 responsiveweb__SliceContainer-sc-2kg7gs-0 c1 css-view-1dbjc4n"
      >
        <responsiveweb__Base>
          <div
            className="c2 responsiveweb__Base-aumpme-1 responsiveweb__Base-aumpme-3 responsiveweb__Base-aumpme-4 c2 css-view-1dbjc4n"
          >
            <responsiveweb__ChildContainer>
              <div
                className="c3 responsiveweb__ChildContainer-aumpme-0 c3 css-view-1dbjc4n"
              >
                <div
                  className="css-text-76zvg2"
                >
                  standard-1
                </div>
              </div>
            </responsiveweb__ChildContainer>
          </div>
        </responsiveweb__Base>
      </div>
    </responsiveweb__SliceContainer>
  </div>
</configweb__Base>
`;

exports[`3. two child elements 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom-color: #DBDBDB;
  border-bottom-width: 1px;
  border-style: solid;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-left: 10px;
  padding-right: 10px;
}

.c4 {
  border-bottom-color: #DBDBDB;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-bottom: 10px;
  margin-top: 10px;
  min-width: auto;
}

.c0 .imageContainerClass {
  display: block;
}

.c0 .summaryHidden {
  display: none;
}

.c0 .summary125Class {
  display: block;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-bottom: 10px;
  padding-top: 10px;
  width: 100%;
}

@media (-webkit-min-device-pixel-ratio:2) {
  .c1 {
    border-bottom-width: 0.5px;
  }
}

@media (min-width:768px) {
  .c3 {
    -webkit-flex-basis: 0 !important;
    -ms-flex-preferred-size: 0 !important;
    flex-basis: 0 !important;
    padding-left: 0;
    padding-right: 0;
  }
}

@media (min-width:768px) {
  .c4 {
    border-bottom: none;
    border-right-color: #DBDBDB;
    border-right-style: solid;
    border-right-width: 1px;
    -webkit-flex: 0 !important;
    -ms-flex: 0 !important;
    flex: 0 !important;
    margin: 0 10px;
  }
}

@media (min-width:768px) {
  .c0 .imageContainerClass {
    display: block;
  }

  .c0 .headlineClass {
    font-size: 25px;
    line-height: 28px;
    margin-top: 5px;
  }
}

@media (min-width:1024px) {
  .c0 .summary125Class {
    display: block;
  }

  .c0 .summary145Class {
    display: none;
  }
}

@media (min-width:768px) {
  .c2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c2 {
    width: 80.8%;
  }
}

@media (min-width:1024px) {
  .c2 {
    width: 56.2%;
  }
}

<configweb__Base>
  <div
    className="c0 configweb__Base-sc-1adgnqc-0 configweb__Base-sc-1adgnqc-2 c0 css-view-1dbjc4n"
  >
    <responsiveweb__SliceContainer>
      <div
        className="c1 responsiveweb__SliceContainer-sc-2kg7gs-0 c1 css-view-1dbjc4n"
      >
        <responsiveweb__Base>
          <div
            className="c2 responsiveweb__Base-aumpme-1 responsiveweb__Base-aumpme-3 responsiveweb__Base-aumpme-4 c2 css-view-1dbjc4n"
          >
            <responsiveweb__ChildContainer>
              <div
                className="c3 responsiveweb__ChildContainer-aumpme-0 c3 css-view-1dbjc4n"
              >
                <div
                  className="css-text-76zvg2"
                >
                  standard-1
                </div>
              </div>
            </responsiveweb__ChildContainer>
            <responsiveweb>
              <div
                className="c4 responsiveweb-sc-2kg7gs-1 c4 css-view-1dbjc4n"
              />
            </responsiveweb>
            <responsiveweb__ChildContainer>
              <div
                className="c3 responsiveweb__ChildContainer-aumpme-0 c3 css-view-1dbjc4n"
              >
                <div
                  className="css-text-76zvg2"
                >
                  standard-2
                </div>
              </div>
            </responsiveweb__ChildContainer>
          </div>
        </responsiveweb__Base>
      </div>
    </responsiveweb__SliceContainer>
  </div>
</configweb__Base>
`;

exports[`4. three child elements 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom-color: #DBDBDB;
  border-bottom-width: 1px;
  border-style: solid;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-left: 10px;
  padding-right: 10px;
}

.c4 {
  border-bottom-color: #DBDBDB;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-bottom: 10px;
  margin-top: 10px;
  min-width: auto;
}

.c0 .imageContainerClass {
  display: none;
}

.c0 .summaryHidden {
  display: none;
}

.c0 .summary125Class {
  display: block;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-bottom: 10px;
  padding-top: 10px;
  width: 100%;
}

@media (-webkit-min-device-pixel-ratio:2) {
  .c1 {
    border-bottom-width: 0.5px;
  }
}

@media (min-width:768px) {
  .c3 {
    -webkit-flex-basis: 0 !important;
    -ms-flex-preferred-size: 0 !important;
    flex-basis: 0 !important;
    padding-left: 0;
    padding-right: 0;
  }
}

@media (min-width:768px) {
  .c4 {
    border-bottom: none;
    border-right-color: #DBDBDB;
    border-right-style: solid;
    border-right-width: 1px;
    -webkit-flex: 0 !important;
    -ms-flex: 0 !important;
    flex: 0 !important;
    margin: 0 10px;
  }
}

@media (min-width:768px) {
  .c0 .imageContainerClass {
    display: block;
  }

  .c0 .headlineClass {
    font-size: 25px;
    line-height: 28px;
    margin-top: 5px;
  }
}

@media (min-width:1024px) {
  .c0 .summary125Class {
    display: none;
  }

  .c0 .summary145Class {
    display: block;
  }
}

@media (min-width:768px) {
  .c2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c2 {
    padding-left: 20px;
    padding-right: 20px;
    width: 100%;
  }
}

@media (min-width:1024px) {
  .c2 {
    width: 100%;
  }
}

<configweb__Base>
  <div
    className="c0 configweb__Base-sc-1adgnqc-0 configweb__Base-sc-1adgnqc-2 c0 css-view-1dbjc4n"
  >
    <responsiveweb__SliceContainer>
      <div
        className="c1 responsiveweb__SliceContainer-sc-2kg7gs-0 c1 css-view-1dbjc4n"
      >
        <responsiveweb__Base>
          <div
            className="c2 responsiveweb__Base-aumpme-1 responsiveweb__Base-aumpme-2 responsiveweb__Base-aumpme-4 c2 css-view-1dbjc4n"
          >
            <responsiveweb__ChildContainer>
              <div
                className="c3 responsiveweb__ChildContainer-aumpme-0 c3 css-view-1dbjc4n"
              >
                <div
                  className="css-text-76zvg2"
                >
                  standard-1
                </div>
              </div>
            </responsiveweb__ChildContainer>
            <responsiveweb>
              <div
                className="c4 responsiveweb-sc-2kg7gs-1 c4 css-view-1dbjc4n"
              />
            </responsiveweb>
            <responsiveweb__ChildContainer>
              <div
                className="c3 responsiveweb__ChildContainer-aumpme-0 c3 css-view-1dbjc4n"
              >
                <div
                  className="css-text-76zvg2"
                >
                  standard-2
                </div>
              </div>
            </responsiveweb__ChildContainer>
            <responsiveweb>
              <div
                className="c4 responsiveweb-sc-2kg7gs-1 c4 css-view-1dbjc4n"
              />
            </responsiveweb>
            <responsiveweb__ChildContainer>
              <div
                className="c3 responsiveweb__ChildContainer-aumpme-0 c3 css-view-1dbjc4n"
              >
                <div
                  className="css-text-76zvg2"
                >
                  standard-3
                </div>
              </div>
            </responsiveweb__ChildContainer>
          </div>
        </responsiveweb__Base>
      </div>
    </responsiveweb__SliceContainer>
  </div>
</configweb__Base>
`;
